onnxruntime/core/providers/cpu/nn/pool_base.h

```diff
@@ -100,190 +100,37 @@ class LpPool {
 
 class PoolBase {
```

Comment:
>PoolBase [](start = 6, length = 8)

Please check other providers using this class:

[https://github.com/microsoft/onnxruntime/search?q=PoolBase&unscoped_q=PoolBase](https://github.com/microsoft/onnxruntime/search?q=PoolBase&unscoped_q=PoolBase)

Reply:
The interface change is more like a simplification to the PoolAttributes class rather than PoolBase class itself. Previously we have something like:

    class A {
        int m_;
        void foo(int m) { /* we use m here; */ }
        void bar() { foo(m_); }
   }

Now we have:

    class A {
      int m_;
      void foo() { /* we use m_ here; */ }
      void bar() { foo(); }
    }

Reply:
I see. Seems the class should avoid using protected members: [https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rh-protected](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rh-protected). For clarity of change history, I'd suggest sending a separate PR for the PoolBase refactoring, and modify all other providers first, then use the updated ones in Nuphar.

---
In reply to: [327817751](https://github.com/microsoft/onnxruntime/pull/1904#discussion_r327817751) [](ancestors = 327817751)

Reply:
Sounds good. I will abandon this PR and create two separate ones. Thanks.